### DISCUSSION TITLE
Working with responsive image

### DISCUSSION THREAD

**Discussion by airdev-web** [spatie/laravel-medialibrary#2513] [Q&A] [answered]
Hello there, greetings from Belgium !

I don't understand how to work with responsive images..

I have created a conversion as on the doc, but impossible to generate the multiple `srcset=""` `<img />` tag with getFirstMedia

``` Php
// My Model
public function registerMediaConversions(Media $media = null): void
{
    $this->addMediaConversion('post_image')
        ->withResponsiveImages();
}
```

``` Php
// My Controller
public function index()
{
    $posts = Post::where('is_active', true)
        ->get();

    $post = $posts[0];
    $media = $post->getFirstMedia('post_image');

    dd($media->toHtml()); // Just to debug what I will have in my blade file

    return view('pages.blog.index', ['posts' => $posts]);
}
```

Here is the ouput
``` html
<img src="http://127.0.0.1:8000/uploads/223/creation-logo-agence-web.jpg" alt="creation-logo-agence-web">
```

As you can see, it's not the multiple `srcset` that should be generated.

Of course I try to use it into my blade view, but it generates me the exact output.
``` Blade
<!-- My Blade view -->
<div>
  {{ $post->getFirstMedia('post_image') }}
</div>
```

Any idea ? 
Thanks from Namur !

**Comment by danielsdeboer** [comment]
Responsive images and other conversions are queued, so make sure you're running a queue worker or they won't be generated.

**Comment by airdev-web** [comment, marked as answer]
I just found the error.

I confused collections and conversions.
The write code is :
```php
$post->getFirstMedia($collection_name)->getUrl($conversion_name)
```
